simpleviewinc/keg-herkin
# Dynamic app urls

## Goal

* Update keg-herkin to infer the client app's URL 
  * we already had access to the keg environment (local vs staging) and the client app's git branch, but we did not have the `alias`, so I added that

## Updates
* `configs/getHerkinConfig.js`
  * refactored `tryRequireSync`
  * fixed bug where the `--config` parameter only worked with paths relative to the keg-herkin directory
* `package.json`
  * add `cli-utils` and fix `axios` vulnerability with a minor update
*  `tasks/definitions/tap/start.js`
  * updated to pass the client app path to `setMountEnvs`
* `tasks/utils/envs/replaceTemplateVars.js`
  * updated to include client app alias in the replace vars
* `tasks/utils/envs/setMountEnvs.js`
  * set default app url to the usual keg format, if the client app didn't override the variable in `world.js`  

## Testing
* `keg herkin && keg pr 49`
* `keg evf && keg br develop && keg pull && keg evf start`
  * this starts a branch of tap-events-force that has a `herkin.config.js`, but no explicit appUrl variable defined
* `keg evf && keg herkin start --from keg-herkin:dynamic-app-urls-from`
* Verify that the herkin app is accessible (should be: http://herkin-dynamic-app-urls.local.keghub.io/)
  * select `openSessionDetailsModal.feature` feature in the editor
* Run the tests for tap-events-force
* Verify that they run and all pass
